corehq/apps/hqwebapp/views.py

```diff
@@ -393,6 +398,43 @@ def logout(req):
 def retrieve_download(req, domain, download_id, template="hqwebapp/file_download.html"):
     return soil_views.retrieve_download(req, download_id, template)
 
+
+def dropbox_next_url(request, download_id):
+    return request.META.get('HTTP_REFERER', '/')
+
+
+@require_dropbox_session(next_url=dropbox_next_url)
+def dropbox_upload(request, download_id):
+    download = DownloadBase.get(download_id)
+    if download is None:
+        logging.error("Download file request for expired/nonexistent file requested")
+        raise Http404
+    else:
+        filename = download.get_filename()
```

Comment:
This needs to use the shared drive since celery and the worker won't be running on the same machine. i don't see logic that does that but maybe it's abstracted underneath this?